snapshot_dbg_cli_tests/test_debuggee_utils.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Unit test file for the debuggee_utils module.
+"""
+
+import copy
+import unittest
+
+from snapshot_dbg_cli.debuggee_utils import get_display_name
+from snapshot_dbg_cli.debuggee_utils import normalize_debuggee
+from snapshot_dbg_cli.debuggee_utils import set_converted_timestamps
+
+
+class SnapshotDebuggerDebuggeeUtilsTests(unittest.TestCase):
+  """ Contains the unit tests for the breakpoint_utils module.
+  """
+
+  def test_set_converted_timestamps(self):
+    """Verifies the set_converted_timestamps() works as expected.
+
+    If the human readable time is not in the breakpoint, but the unix msec
+    version is, the human readable version should be populated, otherwise no
+    changes are done.
+    """
+
+    # Note, these tests don't need fully flesched out breakpoints, the function
+    # under test only cares about time fields in a dict, so these minimal dicts
+    # suffice.
+    testcases = [
+        ('No time fields', {}, {}),
+        (
+            'registrationTime gets populated',
+            {
+                'registrationTimeUnixMsec': 1649962215000
+            },
+            {
+                'registrationTimeUnixMsec': 1649962215000,
+                'registrationTime': '2022-04-14T18:50:15.000000Z'
+            },
+        ),
+        (
+            'registrationTime not overrwritten',
+            {
+                'registrationTimeUnixMsec': 1649962215000,
+                'registrationTime': 'exists not overwritten'
```

Comment:
In what cases will the human readable form be set before conversion?

Reply:
They won't be in practice, but the conversion function was written defensively to not overwrite it if it is set.